core/module.py

```diff
@@ -205,7 +205,11 @@ def __load_status_vars_activate(self, event):
         # add status vars
         for vname, var in self._stat_vars.items():
             sv = self._statusVariables
-            svar = sv[var.name] if var.name in sv else var.default
+            if isinstance(var.default, (OrderedDict, dict)) and var.name in sv:
+                svar = var.default
+                svar.update(sv[var.name])
+            else:
+                svar = sv[var.name] if var.name in sv else var.default
```

Comment:
Just realized that whenever `var.default` is used to initialize the `StatusVar`, a deepcopy of this default value should be assigned. Otherwise the module will alter the default value of the `StatusVar` instance which could lead to undefined behaviour. So please exchange both `var.default` occurrences with `copy.deepcopy(var.default)`.
This has been an oversight before, so let's fix it along with this PR.

Reply:
Thank you for the review. I add this. Theoretical no module read the default value beacause there is a copy in the mata class , but just in case.

Reply:
Well, there is a deepcopy created in the metaclass. However, if you just deactivate and activate the module again, the exact same copy is being used because `ModuleMeta.__new__()` is not being called again. Thus you can construct problematic edge cases from this. And the `StatusVar` default value was definetely the same identity as the exposed value in the module (`var.default is my_module.var_name` evaluates `True`).
Of course under "normal" operation this is not causing any problems but let's improve it anyway.
It is very likely that the deepcopy in `ModuleMeta` can be omitted instead, but since there is a new revision of the entire qudi core underway that changes the module base class and loading procedure quite substantially, I think it's not worth investigating. 
Thank you again.